src/model/Model.hpp

```diff
@@ -526,24 +686,120 @@ namespace model {
   template <>
   MODEL_API OutputControlFiles Model::getUniqueModelObject<OutputControlFiles>();
 
+  template <>
+  MODEL_API OutputControlReportingTolerances Model::getUniqueModelObject<OutputControlReportingTolerances>();
+
+  template <>
+  MODEL_API OutputDiagnostics Model::getUniqueModelObject<OutputDiagnostics>();
+
+  template <>
+  MODEL_API OutputDebuggingData Model::getUniqueModelObject<OutputDebuggingData>();
+
+  template <>
+  MODEL_API OutputJSON Model::getUniqueModelObject<OutputJSON>();
+
+  template <>
+  MODEL_API OutputEnergyManagementSystem Model::getUniqueModelObject<OutputEnergyManagementSystem>();
```

Comment:
I'm going to be annoying here, sorry. Adding cached getters for all unique model objects is nice, but I think we want to put that in a separate PR. I have a very strong feeling that we'll need partial classes adjustments in C#.

Reply:
Moved to https://github.com/NREL/OpenStudio/pull/4604